### PR TITLE
Change SourceFileMapping to allow string or object as the value

### PR DESCRIPTION
**Pull request by pieandcakes**
Change for sourcefilemapping to allow string or an object with string+bool  as the value. 

Also fixed some schema issues and changed description

### CODE REVIEW

**Comment by pieandcakes** [comment]
This is for issue #612

File: src/MIDebugPackage/OpenFolderSchema.json [hunk 1]
```diff
@@ -207,18 +219,35 @@
...
+              "sourceFileMapOptions": {
+                "type": "object",
+                "properties": {
+                  "editorPath": {
+                    "type": "string",
+                    "description": "The location of the source code for the editor to locate."
+                  },
+                  "useForBreakpoints": {
+                    "type": "boolean",
+                    "description": "Use this source mapping for breakpoint binding? Default is true."
```
Comment (gregg-miskelly) [inline src/MIDebugPackage/OpenFolderSchema.json]: >for breakpoint binding [](start = 60, length = 22)

I wonder if people will understand what 'for breakpoint binding' means?

How about:
Use this source mapping when setting breakpoints? Default is 'true'. Set to 'false' if files under 'editorPath' were not always originally compiled from the compiler path. Note that if if a source file doesn't match any entry in the map, the breakpoint will be set using just the file name.

Comment (pieandcakes) [inline src/MIDebugPackage/OpenFolderSchema.json]: @gregg-miskelly Took another stab at it. How about this wording

**Comment by pieandcakes** [comment]
@gregg-miskelly @chuckries Also added newlines in the description text for some of the descriptions so that the default and examples are on a new line.

File: src/RequiredNetFX46FacadeAssemblies.targets [hunk 2]
```diff
@@ -13,5 +13,6 @@
     <RequiredNetFX46FacadeAssemblies Include="$(NuGetPackagesDirectory)\System.Net.Sockets\4.1.0\lib\net46\System.Net.Sockets.dll"/>
     <RequiredNetFX46FacadeAssemblies Include="$(NuGetPackagesDirectory)\System.Security.Cryptography.X509Certificates\4.1.0\lib\net46\System.Security.Cryptography.X509Certificates.dll"/>
     <RequiredNetFX46FacadeAssemblies Include="$(NuGetPackagesDirectory)\System.Runtime.InteropServices.RuntimeInformation\4.0.0\lib\net45\System.Runtime.InteropServices.RuntimeInformation.dll"/>
+    <RequiredNetFX46FacadeAssemblies Include="$(NuGetPackagesDirectory)\System.Runtime.Serialization.Primitives\4.1.1\lib\net46\System.Runtime.Serialization.Primitives.dll"/>
```
Comment (gregg-miskelly) [inline src/RequiredNetFX46FacadeAssemblies.targets]: >RequiredNetFX46FacadeAssemblies [](start = 5, length = 31)

Update src\MIDebugPackage\Install.cmd also

File: src/MIDebugPackage/OpenFolderSchema.json [hunk 3]
```diff
@@ -194,31 +208,48 @@
                   "debuggerPath": {
                     "type": "string",
                     "description": "The full path to the debugger on the target machine, for example /usr/bin/gdb."
-                  },
-                  "pipeEnv": {
+                  },                  "pipeEnv": {
```
Comment (gregg-miskelly) [inline src/MIDebugPackage/OpenFolderSchema.json]: >                   [](start = 20, length = 18)

nit: your spacing looks like it got messed up here

File: src/MIDebugPackage/OpenFolderSchema.json [hunk 4]
```diff
@@ -194,31 +208,48 @@
...
+              "sourceFileMapOptions": {
+                "type": "object",
+                "properties": {
+                  "editorPath": {
+                    "type": "string",
+                    "description": "The location of the source code for the editor to locate."
+                  },
+                  "useForBreakpoints": {
+                    "type": "boolean",
+                    "description": "Use this source mapping when setting breakpoints. If 'false', only the filename and line number is sent. If 'true', only when this mapping applies will the entire path to the file and line number be sent, otherwise just filename and line number is sent. \nDefault is 'true'."
```
Comment (gregg-miskelly) [inline src/MIDebugPackage/OpenFolderSchema.json]: >If 'false', only the filename and line number is sent [](start = 86, length = 53)

Should there be a question mark instead of a period at the end of the first sentence?

RE: "sent" - instead of "only the filename and line number is sent" how about "breakpoints are set using just file name and line number (no path)" since it might not be obvious to people where we are sending this information. Similar comment in the next sentences too.

Last: Do we need a '\n' somewhere in there?

File: src/MIDebugPackage/OpenFolderSchema.json (hunk 3 above)
Comment (pieandcakes) [inline src/MIDebugPackage/OpenFolderSchema.json]: Fixed

File: src/RequiredNetFX46FacadeAssemblies.targets (hunk 2 above)
Comment (pieandcakes) [inline src/RequiredNetFX46FacadeAssemblies.targets]: Fixed

File: src/MIDebugPackage/OpenFolderSchema.json (hunk 4 above)
Comment (pieandcakes) [inline src/MIDebugPackage/OpenFolderSchema.json]: @gregg-miskelly  I'll take another stab at it. I was trying to avoid the first sentence sounding like a question. 
I can add a '\n' after the first sentence.

File: src/MIDebugPackage/OpenFolderSchema.json [hunk 5]
```diff
@@ -204,21 +218,38 @@
...
+              "sourceFileMapOptions": {
+                "type": "object",
+                "properties": {
+                  "editorPath": {
+                    "type": "string",
+                    "description": "The location of the source code for the editor to locate."
+                  },
+                  "useForBreakpoints": {
+                    "type": "boolean",
+                    "description": "When setting breakpoints, this source mapping should be used. \nIf 'false', only the filename and line number is used for setting breakpoints. \nIf 'true', breakpoints will be set with the full path to the file and line number only when this source mapping is used. Otherwise just filename and line number will be used when setting breakpoints. \nDefault is 'true'."
```
Comment (pieandcakes) [inline src/MIDebugPackage/OpenFolderSchema.json]: @gregg-miskelly does this sound better:
>"When setting breakpoints, this source mapping should be used. \nIf 'false', only the filename and line number is used for setting breakpoints. \nIf 'true', breakpoints will be set with the full path to the file and line number only when this source mapping is used. Otherwise just filename and line number will be used when setting breakpoints. \nDefault is 'true'."

**Comment by pieandcakes** [comment]
@gregg-miskelly Removed the changes for fixing the missing DLL issue. We're going to make a different change and I didn't want to leave it in this bug.

**Comment by gregg-miskelly** [comment]
Okay, merge at will...
